### ISSUE TITLE
react/no-unstable-nested-components does not detect components wrapped in Array.map()

### ISSUE THREAD

**Issue by gusttav-lang** [jsx-eslint/eslint-plugin-react#3492]
This component is not detected:

```
export default function Component({ title, dependentComponents }) {
  function InternalComponent() {
    // uncomment next 2 lines and comment Array.map line to trigger the rule
    // const index = 0;
    // const component = dependentComponents[index];

    return dependentComponents.map((component, index) => {
      return (
        <div key={index}>
          <h1>{component.title}</h1>
          <p>{component.description}</p>
        </div>
      );
    });
  }

  return (
    <div>
      <h1>{title}</h1>
      <InternalComponent />
    </div>
  );
}
```

https://stackblitz.com/edit/nextjs-dxtyfp?file=pages/index.js

**Comment by akulsr0** [comment]
The `InternalComponent` is already getting warning, and since array map callbacks aren't component no warning is coming there. It seems working as expected to me.
![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/43666833/f3f5e759-fe6d-441e-b3a8-57aa60f12669)

**Comment by akulsr0** [comment]
@ljharb This seems working as expected to me, can be closed.
